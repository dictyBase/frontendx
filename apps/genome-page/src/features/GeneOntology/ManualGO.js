// @flow
import React from "react"
import { connect } from "react-redux"
import DisplayTable from "./DisplayTable"
import WithGoDataRendering from "./WithGoDataRendering"
import { Wrapper, GoHeaderStyle } from "styles/style"

type Props = {
  /** The data fetched from the API */
  goData: Object,
}

/**
 * Filters and shares only the Manual GO data (IMP, IGI, IDA, IBA)
 */

export const ManualGO = (props: Props) => {
  const data = props.goData.data.filter(
    code =>
      code.evidence === "IMP" ||
      code.evidence === "IGI" ||
      code.evidence === "IDA" ||
      code.evidence === "IBA",
  )
  return (
    <Wrapper>
      <GoHeaderStyle>
        <h3>Molecular Function</h3>
      </GoHeaderStyle>
      <DisplayTable data={data} />
      <br />
      <GoHeaderStyle>
        <h3>Biological Process</h3>
      </GoHeaderStyle>
      <DisplayTable data={data} />
      <br />
      <GoHeaderStyle>
        <h3>Cellular Composition</h3>
      </GoHeaderStyle>
      <DisplayTable data={data} />
    </Wrapper>
  )
}

const mapStateToProps = ({ goData }) => ({ goData })

export default connect(mapStateToProps, null)(WithGoDataRendering(ManualGO))
