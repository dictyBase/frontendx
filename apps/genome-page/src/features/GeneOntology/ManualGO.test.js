import React from "react"
import renderer from "react-test-renderer"
import { ManualGO } from "./ManualGO"
jest.mock("react-virtualized")

test("matching a snapshot of ManualGO", () => {
  const goData = {
    data: [
      {
        term: "ATP-gated ion channel activity",
        evidence: "IMP",
        with: "",
        reference: "Sivaramakrishnan & Fountain (2012)",
        date: "21-02-2014",
        source: "DDB",
      },
      {
        term: "calcium ion transmembrane transport requires ATP",
        evidence: "IGI",
        with: "p2xA p2xC p2xD p2xE",
        reference: "Sivaramakrishnan & Fountain (2012)",
        date: "21-02-2014",
        source: "DDB",
      },
    ],
  }
  const fetchData = () => {}

  const component = renderer.create(
    <ManualGO goData={goData} fetchData={fetchData} />,
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
