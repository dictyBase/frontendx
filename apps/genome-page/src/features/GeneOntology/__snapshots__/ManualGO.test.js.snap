// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matching a snapshot of ManualGO 1`] = `
<div
  className="sc-eilVRo fSGauw"
>
  <div
    className="sc-eerKOB kCQXGC"
  >
    <h3>
      Molecular Function
    </h3>
  </div>
  <div
    className="wrapper"
  />
  <br />
  <div
    className="sc-eerKOB kCQXGC"
  >
    <h3>
      Biological Process
    </h3>
  </div>
  <div
    className="wrapper"
  />
  <br />
  <div
    className="sc-eerKOB kCQXGC"
  >
    <h3>
      Cellular Composition
    </h3>
  </div>
  <div
    className="wrapper"
  />
</div>
`;
